core-v1/src/main/scala/engine/SimpleEngine.scala

```diff
@@ -0,0 +1,83 @@
+package com.rallyhealth.vapors.v1
+
+package engine
+
+import algebra.Expr
+import data.FactTable
+
+import cats.Foldable
+
+/**
+  * A vapors [[Expr]] interpreter that just builds a simple function without providing any post-processing.
+  *
+  * This can be used for better performance when you do not need to inspect or serialize the intermediate
+  * results of computation.
+  */
+object SimpleEngine {
+
+  @inline def apply[OP[_]](factTable: FactTable): Visitor[OP] = new Visitor(factTable)
+
+  class Visitor[OP[_]](protected val factTable: FactTable) extends Expr.Visitor[Lambda[(`-I`, `+O`) => I => O], OP] {
+
+    import cats.implicits._
+
+    override def visitAnd[I](expr: Expr.And[I, OP])(implicit opO: OP[Boolean]): I => Boolean = { i =>
+      expr.leftExpr.visit(this)(i) && expr.rightExpr.visit(this)(i)
+    }
+
+    override def visitAndThen[II, IO : OP, OI, OO : OP](
+      expr: Expr.AndThen[II, IO, OI, OO, OP],
+    )(implicit
+      evBI: IO <:< OI,
+    ): II => OO = { ii =>
+      val io = expr.inputExpr.visit(this)(ii)
+      val oo = expr.outputExpr.visit(this)(io)
+      oo
+    }
```

Comment:
The `SimpleEngine` does not call the attached debugger, but the `StandardEngine` does... I should either remove debugging or make it call it here.

Reply:
I make this change in a later commit that is bound up with adding unit tests for this very change. These unit tests depend on a LOT of other changes that I don't want to bring in... so I think I'm gonna skip this comment for now. It will be in the [next PR](https://github.com/jeffmay/vapors/pull/68/files#diff-0eb9ebcb364cc98bd88c65e0bf921f426b8e746c6d5a0ad92fbc203160620562R45)